### ISSUE TITLE
Http Request Method : Add Basic Authentification

### ISSUE THREAD

**Issue by jmberhault** [Yadoms/yadoms#635]
For plugins, to completly integrate Http request developed into yadoms/shared, we must integrate the basic authentification (_please check WES plugin to see an example of implementation. This one could also be optimized, if credentials are needed, we could send directly Basic Auth, to avoid the double frame to the equipment_).

**Comment by sgallou** [comment]
We probably should refactor CHttpMethods static methods, to add a CHttpRequest class that can handle all needs of a Http request (handle header params, basic authent, standard/secured session, specific timeout, handle JSON or others response formats...), in a user-friendly form (user must set only needed parameters for its usage).

**Comment by oben01** [comment]
@sgallou @jmberhault Basic authentification could be done by setting the Authorization header as following : 

```c++
   std::string username = ''SUPER-JM";
   std::string password = ''MY STRONG PASSWORD";

   const std::string authorizationType = "Basic ";
   const auto authorizationHeader = username + ":" + password;

   std::map<std::string, std::string> headerParameters;
   headerParameters["Authorization"] = authorizationType + shared::encryption::CBase64::encode(
                                                                   reinterpret_cast<const unsigned char*>(authorizationHeader.c_str()),
                                                                   authorizationHeader.length());
   
   std::string url = "YOUR/URL";
   shared::CDataContainer body;
   try
   {
      shared::http::CHttpMethods::sendPostRequest(url,
                                                  body.serialize(),
                                                  headerParameters);
   }
   catch (std::exception& e)
   {
      const auto message = (boost::format("Fail to send Post http request or interpret answer \"%1%\" : %2%") % url %
         e.what()).str();
      YADOMS_LOG(error) << "Fail to send Post http request or interpret answer " << url << " : " << e.what();
   }

```

**Comment by sgallou** [comment]
Closed by e1e66e9cb35e2b084b4660382348e84fa45d3634

@oben01 Calling curlpp.setOpt(curlpp::options::UserPwd("user", "passwd")) was enough and simpler

**Comment by oben01** [comment]
we decided to not abstract the use of curlpp.
@jmberhault you can use dirrectly the createHttpRestRequest method from the shared.
A simple example could be :

```c++
      boost::shared_ptr<shared::CDataContainer> response;
      shared::http::CHttpRestHelpers::createHttpRestRequest(shared::http::CCurlppHttpRestRequest::EType::kPost, url)
         ->withBody(body)
         .withHeaderParameters(headerPostParameters)
         .withBasicAuthentication("Basic", "MYPASSWORD")
         .send([&response](auto data)
         {
            response = data;
         });
```
